testsuite/cluster-test/src/main.rs

```diff
@@ -342,30 +352,69 @@ impl BasicSwarmUtil {
 
 impl ClusterUtil {
     pub fn setup(args: &Args) -> Self {
-        let aws = Aws::new();
-        let cluster = Cluster::discover(&aws).expect("Failed to discover cluster");
-        let cluster = if args.peers.is_empty() {
-            cluster
-        } else {
-            cluster.validator_sub_cluster(args.peers.clone())
-        };
-        let prometheus = Prometheus::new(
-            cluster
-                .prometheus_ip()
-                .expect("Failed to discover prometheus ip in aws"),
-            aws.workspace(),
-        );
-        info!(
-            "Discovered {} validators and {} fns in {} workspace",
-            cluster.validator_instances().len(),
-            cluster.fullnode_instances().len(),
-            aws.workspace()
-        );
-        Self {
-            cluster,
-            aws,
-            prometheus,
-        }
+        Runtime::new().unwrap().block_on(async move {
+            let k8s = env::var("KUBERNETES_SERVICE_HOST").is_ok();
+            let aws = Aws::new(k8s);
+            let cluster_swarm = ClusterSwarmKube::new()
+                .await
+                .expect("Failed to initialize ClusterSwarmKube");
+            let cluster = if k8s {
+                cluster_swarm.delete_all().await.expect("delete_all failed");
+                info!("Successfully delete all old pods");
+                let image_tag = if let Some(image) = args.deploy.as_ref() {
+                    image.clone()
+                } else {
+                    "master".to_string()
+                };
+                info!(
+                    "Deploying with {} tag for validators and fullnodes",
+                    image_tag
+                );
+                cluster_swarm
+                    .create_validator_and_fullnode_set(
+                        args.k8s_num_validators,
+                        args.k8s_fullnodes_per_validator,
+                        &image_tag,
+                        true,
+                    )
+                    .await
+                    .expect("Failed to create_validator_and_fullnode_set");
+                info!("Deployment complete");
+                Cluster::k8s(
+                    cluster_swarm.validator_instances().await,
+                    cluster_swarm.fullnode_instances().await,
+                )
+                .unwrap()
+            } else {
+                Cluster::discover(&aws).expect("Failed to discover cluster")
+            };
+            let cluster = if args.peers.is_empty() {
+                cluster
+            } else {
+                cluster.validator_sub_cluster(args.peers.clone())
+            };
+            let prometheus_ip = if k8s {
+                "libra-testnet-prometheus-server.default.svc.cluster.local"
```

Comment:
We do share prometheus URL in logs so that ppl can look at it, but this URL does not seem to be shareable?

Reply:
This is only an internal URL which will be used by code. For external sharing, the url looks somthing like http://grafana.kush-k8s-testnet.aws.hlw3truzy4ls.com/ - I'll add this in a later diff because it requires quering the workspace from within k8s

Reply:
Updated code so that we print correct public url in logs